### PR TITLE
Implementing flashcardService

### PR DESCRIPTION
**Pull request by Keyshal-211**
## Description of changes
Please provide a short description of the change or if the change is sufficiently described in the GitHub issue, provide a link to the issue.

  
## How has this been tested?
Please describe the test strategy you followed.

- [ ] automated unit test
- [ ] automated integration test
- [ ] automated acceptance test
- [ ] manual, exploratory test

In case of manual test, please document the test well including a set of user instructions and prerequisites. Each including an action, it's result, and where appropriate a screenshot.
In case of automated tests, the test coverage should be reasonably high, especially on Service classes and other classes with complex logic.
    
## Checklist before requesting a review
- [x] My code follows the [coding guidelines](https://github.com/IT-REX-Platform/wiki/blob/main/dev-manuals/backend/coding-guidelines.md) of this project
- [ ] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My code fulfills all acceptance criteria
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] The test coverage (line and branch) is reasonably high, especially on Service classes and other classes with complex logic
- [ ] I have made corresponding changes to the documentation
- [ ] I have added explanation of architectural decision and rationales to [wiki/adr](https://github.com/IT-REX-Platform/wiki/tree/main/adr)
- [ ] I have updated the changes in the ticket description

## Checklist for reviewer
- [ ] The code works and does not throw errors
- [ ] The code is easy to understand and there are no confusing parts
- [x ] The code follows the [coding guidelines](https://github.com/IT-REX-Platform/wiki/blob/main/dev-manuals/backend/coding-guidelines.md) of this project
- [ ] The code change accomplishes what it is supposed to do
- [ ] I cannot think of any use case in which the code does not behave as intended
- [ ] The added and existing tests reasonably cover the code change
- [ ] I cannot think of any test cases, input or edge cases that should be tested in addition
- [ ] Description of the change is included in the documentation

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -1,74 +1,48 @@
-version: "3"
```
Comment (PaulBredl) [inline docker-compose.yml:1]: Ports 3000, 3001 and so on are already occupied by the media serivce. Please use port 5000 and onwards.

File: src/main/java/de/unistuttgart/iste/gits/flashcardservice/FlashcardServiceApplication.java [hunk 2]
```diff
@@ -1,4 +1,4 @@
-package de.unistuttgart.iste.gits.template;
+package de.unistuttgart.iste.gits.flashcardservice;
```
Comment (PaulBredl) [inline src/main/java/de/unistuttgart/iste/gits/flashcardservice/FlashcardServiceApplication.java]: Please rename the package to "flashcard_service"

File: src/main/java/de/unistuttgart/iste/gits/flashcardservice/config/package-info.java [hunk 3]
```diff
@@ -1,4 +1,4 @@
 /**
  * This package should contain any classes that are used to configure the application.
  */
-package de.unistuttgart.iste.gits.template.config;
```
Comment (PaulBredl) [inline src/main/java/de/unistuttgart/iste/gits/flashcardservice/config/package-info.java]: Please remove all the package info files from the template

File: src/main/java/de/unistuttgart/iste/gits/flashcardservice/controller/FlashcardController.java [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+
+@Slf4j
+@Controller
+@RequiredArgsConstructor
+public class FlashcardController {
+
+    private final FlashcardService flashcardService;
+
+
+    public FlashcardController(FlashcardService flashcardService) {
```
Comment (PaulBredl) [inline src/main/java/de/unistuttgart/iste/gits/flashcardservice/controller/FlashcardController.java]: Remove this duplicate constructor. This is already defined using `@RequiedArgsConstructor`

File: src/main/java/de/unistuttgart/iste/gits/flashcardservice/controller/FlashcardController.java [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+        this.flashcardService = flashcardService;
+    }
+    @QueryMapping
```
Comment (PaulBredl) [inline src/main/java/de/unistuttgart/iste/gits/flashcardservice/controller/FlashcardController.java]: Formatting is not correct. Please format the code with "Ctrl+L" (on Windows at least). Alternatively there is an option in the commit window, that will automatically format all files correctly

File: src/main/java/de/unistuttgart/iste/gits/flashcardservice/controller/FlashcardController.java [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+    public List<Flashcard> flashcardByIds(@Argument(name = "ids") List<Long> ids) {
+        log.info("Request flashcard by Ids");
+        return flashcardService.getFlashcardById(ids);
+    }
+    @QueryMapping
+    public List<FlashcardSet> flashcardSetsByIds(@Argument(name = "ids") List<Long> ids) {
+        log.info("Request flashcard by Ids");
```
Comment (PaulBredl) [inline src/main/java/de/unistuttgart/iste/gits/flashcardservice/controller/FlashcardController.java]: Either you log all requests (also those below) or none of them. I would personally remove the log statements.

File: src/main/java/de/unistuttgart/iste/gits/flashcardservice/controller/FlashcardController.java [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+        return flashcardService.getFlashcardSetsById(ids);
+    }
+    @QueryMapping
+    public List<FlashcardSet> flashcardSetsByAssessmentIds(@Argument(name = "ids") List<Long> ids) {
```
Comment (PaulBredl) [inline src/main/java/de/unistuttgart/iste/gits/flashcardservice/controller/FlashcardController.java]: Wrong argument: Must be a list of UUIDs (we don't use Longs)

File: src/main/java/de/unistuttgart/iste/gits/flashcardservice/controller/FlashcardController.java [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+    public Flashcard createFlashcard(@Argument(name = "input") CreateFlashcardInput input) {
+        return flashcardService.createFlashcard(input);
+    }
+
+    @MutationMapping
+    public Flashcard updateFlashcard(@Argument(name = "input") UpdateFlashcardInput input) {
+        return flashcardService.updateFlashcard(input);
+    }
+    @MutationMapping
+    public Long deleteFlashcard(@Argument(name = "id") Long id) {
```
Comment (PaulBredl) [inline src/main/java/de/unistuttgart/iste/gits/flashcardservice/controller/FlashcardController.java]: Also here: We use UUIDs, not longs

File: src/main/java/de/unistuttgart/iste/gits/flashcardservice/controller/FlashcardController.java [hunk 9]
```diff
@@ -0,0 +1,68 @@
...
+        return flashcardService.deleteFlashcard(id);
+    }
+
+    @MutationMapping
+    public Flashcard createFlashcardSet(@Argument(name = "input") CreateFlashcardSetInput input) {
+        return flashcardService.createFlashcardSet(input);
+    }
+    @MutationMapping
+    public Long deleteFlashcardSet(@Argument(name = "id") Long id) {
```
Comment (PaulBredl) [inline src/main/java/de/unistuttgart/iste/gits/flashcardservice/controller/FlashcardController.java]: And here same as above

File: src/main/java/de/unistuttgart/iste/gits/flashcardservice/persistence/dao/FlashcardEntity.java [hunk 10]
```diff
@@ -1,17 +1,17 @@
...
+@Entity(name = "Flashcard")
 @Data
 @Builder
 @NoArgsConstructor
 @AllArgsConstructor
-public class TemplateEntity {
+public class FlashcardEntity {
 
     @Id
     @GeneratedValue(strategy = GenerationType.AUTO)
```
Comment (PaulBredl) [inline src/main/java/de/unistuttgart/iste/gits/flashcardservice/persistence/dao/FlashcardEntity.java]: UUIDs instead of Long

Comment (PaulBredl) [inline src/main/java/de/unistuttgart/iste/gits/flashcardservice/persistence/dao/FlashcardEntity.java]: Where are the fields of a the flashcard? Did you forget to commit something? Why is it empty?

File: src/main/java/de/unistuttgart/iste/gits/flashcardservice/persistence/mapper/FlashcardMapper.java [hunk 11]
```diff
@@ -0,0 +1,31 @@
...
+import de.unistuttgart.iste.gits.flashcardservice.persistence.dao.FlashcardEntity;
+import lombok.RequiredArgsConstructor;
+import org.modelmapper.ModelMapper;
+import org.springframework.stereotype.Component;
+@Component
+@RequiredArgsConstructor
+public class FlashcardMapper {
+
+    private final ModelMapper modelMapper;
+    public FlashcardMapper(ModelMapper modelMapper) {
```
Comment (PaulBredl) [inline src/main/java/de/unistuttgart/iste/gits/flashcardservice/persistence/mapper/FlashcardMapper.java]: Also here please remove the duplicated constructor and format the file correctly

File: src/main/java/de/unistuttgart/iste/gits/flashcardservice/persistence/repository/FlashcardRepository.java [hunk 12]
```diff
@@ -0,0 +1,20 @@
...
+
+import java.util.List;
+
+
+
+/**
+ * Repository for {@link FlashcardEntity}.
+ */
+@Repository
+public interface FlashcardRepository extends JpaRepository<FlashcardEntity, Long>, JpaSpecificationExecutor<FlashcardEntity> {
```
Comment (PaulBredl) [inline src/main/java/de/unistuttgart/iste/gits/flashcardservice/persistence/repository/FlashcardRepository.java]: UUID instead of Long

File: src/main/java/de/unistuttgart/iste/gits/flashcardservice/service/FlashcardService.java [hunk 13]
```diff
@@ -0,0 +1,72 @@
...
+    }
+    public Flashcard updateFlashcard(UpdateFlashcardInput input) {
+        flashcardValidator.validateUpdateFlashcardInput(input);
+        requireFlashcardExisting(input.getId());
+
+        FlashcardEntity updatedFlashcardEntity = flashcardRepository.save(flashcardMapper.dtoToEntity(input));
+
+        return flashcardMapper.entityToDto(updatedFlashcardEntity);
+    }
+    public Long deleteFlashcard(Long long1) {
```
Comment (PaulBredl) [inline src/main/java/de/unistuttgart/iste/gits/flashcardservice/service/FlashcardService.java]: UUID instead of Long, please change it everywhere

File: src/main/java/de/unistuttgart/iste/gits/flashcardservice/service/FlashcardService.java [hunk 14]
```diff
@@ -0,0 +1,72 @@
...
+    public void requireFlashcardExisting(Long id) {
+        if (!flashcardRepository.existsById(id)) {
+            throw new EntityNotFoundException("Flashcard with id " + id + " not found");
+        }
+    }
+    public List<Flashcard> getFlashcardById(List<Long> ids) {
+        return flashcardRepository.findById(ids).stream().map(flashcardMapper::entityToDto).toList();
+    }
+    public List<FlashcardSet> getFlashcardSetsById(List<Long> ids) {
+        return flashcardRepository.findById(ids).stream().map(flashcardMapper::entityToDto).toList();
```
Comment (PaulBredl) [inline src/main/java/de/unistuttgart/iste/gits/flashcardservice/service/FlashcardService.java]: Here you query a flashcard, but a flashcardSet is required

File: src/main/java/de/unistuttgart/iste/gits/flashcardservice/service/FlashcardService.java [hunk 15]
```diff
@@ -0,0 +1,72 @@
...
+    }
+
+    public List<FlashcardSet> getFlashcardSetsByAssessmentId(List<Long> ids) {
+        return flashcardRepository.findById(ids).stream().map(flashcardMapper::entityToDto).toList();
```
Comment (PaulBredl) [inline src/main/java/de/unistuttgart/iste/gits/flashcardservice/service/FlashcardService.java]: Also here you query flashcards instead of flashcard sets?

File: src/main/java/de/unistuttgart/iste/gits/flashcardservice/validation/FlashcardValidator.java [hunk 16]
```diff
@@ -0,0 +1,21 @@
+package de.unistuttgart.iste.gits.flashcardservice.validation;
+import org.springframework.stereotype.Component;
+
+@Component
+
```
Comment (PaulBredl) [inline src/main/java/de/unistuttgart/iste/gits/flashcardservice/validation/FlashcardValidator.java]: Formatting

File: src/main/java/de/unistuttgart/iste/gits/flashcardservice/validation/FlashcardValidator.java [hunk 17]
```diff
@@ -0,0 +1,21 @@
...
+public class FlashcardValidator {
+
+    public void validateCreateFlashcardInput(CreateFlashcardInput flashcardInput) {
+        // TODO implement validation
```
Comment (PaulBredl) [inline src/main/java/de/unistuttgart/iste/gits/flashcardservice/validation/FlashcardValidator.java]: Is there nothing to validate?

File: docker-compose.yml [hunk 18]
```diff
@@ -1,74 +1,48 @@
...
-      - "1032:5432"
+      - "3032:5432"
     volumes:
-      - templatedata:/var/lib/postgresql/data
+      - flashcarddata:/var/lib/postgresql/data
     environment:
       - POSTGRES_USER=root
       - POSTGRES_PASSWORD=root
-      - POSTGRES_DB=templatedatabase
+      - POSTGRES_DB=flashcard_service
```
Comment (PaulBredl) [inline docker-compose.yml:15]: here you call it "flashcard_service" but in the application properties you call it "flashcardservice" This wont work, you need to name it the same

File: docker-compose.yml [hunk 19]
```diff
@@ -1,74 +1,48 @@
...
   testDb:
     image: postgres:alpine
     restart: always
     expose:
-      - 1033
+      - 4033
```
Comment (PaulBredl) [inline docker-compose.yml]: Also port 50..+

File: src/main/resources/application-dev.properties [hunk 20]
```diff
@@ -1,7 +1,7 @@
 # Database
 spring.jpa.properties.hibernate.dialect=org.hibernate.dialect.PostgreSQLDialect
 #TODO set port and name for service database
-spring.datasource.url=jdbc:postgresql://localhost:1032/templatedatabase
+spring.datasource.url=jdbc:postgresql://localhost:1032/flashcarddatabase
```
Comment (PaulBredl) [inline src/main/resources/application-dev.properties]: Inconsistent name with docker file

File: src/main/resources/application.properties [hunk 21]
```diff
@@ -1,10 +1,20 @@
...
-# TODO set port
-server.port=1001
-dapr.appId=template
-dapr.port=1000
\ No newline at end of file
+server.port=3001
+dapr.appId=flashcard-service
+dapr.port=3000
+spring.jpa.properties.hibernate.dialect=org.hibernate.dialect.PostgreSQLDialect
+spring.datasource.url=jdbc:postgresql://localhost:1032/flashcard_service
```
Comment (PaulBredl) [inline src/main/resources/application.properties]: Here ist is correct though!

File: src/main/resources/application.properties [hunk 22]
```diff
@@ -1,10 +1,20 @@
-# for deployment or when using docker compose = prod, for local development = dev
```
Comment (PaulBredl) [inline src/main/resources/application.properties]: Edit: Please do not remove this first line `spring.profiles.active=prod`

File: src/main/resources/application.properties [hunk 23]
```diff
@@ -1,10 +1,20 @@
...
 # enable graphiql (graphiql is a web interface for exploring GraphQL)
 spring.graphql.graphiql.enabled=true
 spring.graphql.graphiql.path=/graphiql
 
-# TODO set port
-server.port=1001
-dapr.appId=template
-dapr.port=1000
\ No newline at end of file
+server.port=3001
```
Comment (PaulBredl) [inline src/main/resources/application.properties]: Port needs to be updated

**Review by PaulBredl — CHANGES_REQUESTED** [review]
Sorry, I have quite a lot to dos for you 😅 But it's a good starting point.
Please make sure that the service runs and try out the queries to see if it all works.

**Comment by Keyshal-211** [comment]
I build the project, it is successfully build, but as you said I will change all ports and do what you write down in the comment

File: docker-compose.yml (hunk 1 above)
Comment (Keyshal-211) [reply docker-compose.yml:1]: done

File: src/main/java/de/unistuttgart/iste/gits/flashcardservice/FlashcardServiceApplication.java (hunk 2 above)
Comment (Keyshal-211) [reply src/main/java/de/unistuttgart/iste/gits/flashcardservice/FlashcardServiceApplication.java]: DONE

Comment (Keyshal-211) [reply src/main/java/de/unistuttgart/iste/gits/flashcardservice/FlashcardServiceApplication.java]: DONE

File: src/main/java/de/unistuttgart/iste/gits/flashcardservice/config/package-info.java (hunk 3 above)
Comment (Keyshal-211) [reply src/main/java/de/unistuttgart/iste/gits/flashcardservice/config/package-info.java]: DONE

File: src/main/java/de/unistuttgart/iste/gits/flashcardservice/controller/FlashcardController.java (hunk 5 above)
Comment (Keyshal-211) [reply src/main/java/de/unistuttgart/iste/gits/flashcardservice/controller/FlashcardController.java]: done

File: src/main/java/de/unistuttgart/iste/gits/flashcardservice/controller/FlashcardController.java (hunk 7 above)
Comment (Keyshal-211) [reply src/main/java/de/unistuttgart/iste/gits/flashcardservice/controller/FlashcardController.java]: done

File: src/main/java/de/unistuttgart/iste/gits/flashcardservice/service/FlashcardService.java (hunk 15 above)
Comment (Keyshal-211) [reply src/main/java/de/unistuttgart/iste/gits/flashcardservice/service/FlashcardService.java]: did flashcardset to flashcard

File: src/main/java/de/unistuttgart/iste/gits/flashcardservice/validation/FlashcardValidator.java (hunk 17 above)
Comment (Keyshal-211) [reply src/main/java/de/unistuttgart/iste/gits/flashcardservice/validation/FlashcardValidator.java]: I have to do this, think about this and discuss with you

File: src/main/resources/application-dev.properties (hunk 20 above)
Comment (Keyshal-211) [reply src/main/resources/application-dev.properties:4]: "flashcard-database" I did that instead of that

Comment (Keyshal-211) [reply src/main/resources/application-dev.properties]: flashcard_database

File: docker-compose.yml [hunk 24]
```diff
@@ -1,74 +1,75 @@
...
 volumes:
-  templatedata:
+  flashcarddata:
   testdata:
-networks:
-  default:
-    name: dapr-network
-    external: true
\ No newline at end of file
+  #networks:
```
Comment (PaulBredl) [inline docker-compose.yml]: this network stuff needs to be there but I don't know the details

File: src/main/java/de/unistuttgart/iste/gits/flashcard_service/FlashcardServiceApplication.java [hunk 25]
```diff
@@ -8,16 +8,14 @@
...
- * <p>
- * TODO: Rename the package and the class to match the microservice name.
  */
 @SpringBootApplication
 @Slf4j
-public class TemplateForMicroservicesApplication {
+public class FlashcardServiceApplication {
 
     public static void main(String[] args) {
         Arrays.stream(args).map(arg -> "Received argument: " + arg).forEach(log::info);
```
Comment (PaulBredl) [inline src/main/java/de/unistuttgart/iste/gits/flashcard_service/FlashcardServiceApplication.java]: TODO: remove log statement (not needed)

File: src/main/java/de/unistuttgart/iste/gits/flashcard_service/controller/FlashcardController.java [hunk 26]
```diff
@@ -0,0 +1,64 @@
...
+import org.springframework.graphql.data.method.annotation.QueryMapping;
+import org.springframework.stereotype.Controller;
+
+import java.util.List;
+import java.util.UUID;
+
+
+@Slf4j
+@Controller
+
```
Comment (PaulBredl) [inline src/main/java/de/unistuttgart/iste/gits/flashcard_service/controller/FlashcardController.java]: TODO: formatting is wrong here

File: src/main/java/de/unistuttgart/iste/gits/flashcard_service/persistence/dao/FlashcardSideEntity.java [hunk 27]
```diff
@@ -0,0 +1,35 @@
...
+    private String label;
+
+    @Column(nullable = false)
+    private boolean isQuestion;
+
+    @ManyToOne
+    @ToString.Exclude
+    @EqualsAndHashCode.Exclude
+    private FlashcardEntity flashcard;
+
```
Comment (PaulBredl) [inline src/main/java/de/unistuttgart/iste/gits/flashcard_service/persistence/dao/FlashcardSideEntity.java]: TODO remove empty lines

File: src/main/java/de/unistuttgart/iste/gits/flashcard_service/persistence/mapper/FlashcardMapper.java [hunk 28]
```diff
@@ -0,0 +1,54 @@
...
+
+import de.unistuttgart.iste.gits.flashcard_service.persistence.dao.FlashcardEntity;
+import de.unistuttgart.iste.gits.flashcard_service.persistence.dao.FlashcardSetEntity;
+import de.unistuttgart.iste.gits.flashcard_service.persistence.dao.FlashcardSideEntity;
+import de.unistuttgart.iste.gits.generated.dto.*;
+import org.modelmapper.ModelMapper;
+import org.springframework.stereotype.Component;
+
+@Component
+
```
Comment (PaulBredl) [inline src/main/java/de/unistuttgart/iste/gits/flashcard_service/persistence/mapper/FlashcardMapper.java]: formatting

File: src/main/java/de/unistuttgart/iste/gits/flashcard_service/service/FlashcardService.java [hunk 29]
```diff
@@ -0,0 +1,118 @@
...
+    private final FlashcardRepository flashcardRepository;
+    private final FlashcardSetRepository flashcardSetRepository;
+    private final FlashcardMapper flashcardMapper;
+    private final FlashcardValidator flashcardValidator;
+
+
+    public Flashcard createFlashcard(CreateFlashcardInput flashcardInput) {
+        flashcardValidator.validateCreateFlashcardInput(flashcardInput);
+
+        FlashcardEntity mappedEntity = flashcardMapper.dtoToEntity(flashcardInput);
```
Comment (PaulBredl) [inline src/main/java/de/unistuttgart/iste/gits/flashcard_service/service/FlashcardService.java:35]: TODO in this class: there is duplication in the  mapping logic. This should be extracted in methods or maybe done in the mapper, not in the service itsself.

File: src/main/java/de/unistuttgart/iste/gits/flashcard_service/validation/FlashcardValidator.java [hunk 30]
```diff
@@ -0,0 +1,25 @@
...
+import de.unistuttgart.iste.gits.generated.dto.CreateFlashcardInput;
+import de.unistuttgart.iste.gits.generated.dto.CreateFlashcardSetInput;
+import de.unistuttgart.iste.gits.generated.dto.UpdateFlashcardInput;
+import org.springframework.stereotype.Component;
+
+@Component
+
+public class FlashcardValidator {
+
+    public void validateCreateFlashcardInput(CreateFlashcardInput flashcardInput) {
```
Comment (PaulBredl) [inline src/main/java/de/unistuttgart/iste/gits/flashcard_service/validation/FlashcardValidator.java:12]: TODO: implement validation logic

Comment (PaulBredl) [reply src/main/java/de/unistuttgart/iste/gits/flashcard_service/validation/FlashcardValidator.java:12]: for all three methods

File: src/test/java/de/unistuttgart/iste/gits/flashcard_service/FlashcardServiceApiTests.java [hunk 31]
```diff
@@ -0,0 +1,12 @@
+package de.unistuttgart.iste.gits.flashcard_service;
```
Comment (PaulBredl) [inline src/test/java/de/unistuttgart/iste/gits/flashcard_service/FlashcardServiceApiTests.java:1]: TODO: add real api tests

File: src/main/java/de/unistuttgart/iste/gits/flashcard_service/service/FlashcardService.java [hunk 32]
```diff
@@ -0,0 +1,118 @@
...
+        requireFlashcardExisting(uuid);
+        flashcardRepository.deleteById(uuid);
+        return uuid;
+    }
+
+    public FlashcardSet createFlashcardSet(CreateFlashcardSetInput flashcardSetInput) {
+        flashcardValidator.validateCreateFlashcardSetInput(flashcardSetInput);
+
+        FlashcardSetEntity mappedEntity = flashcardMapper.flashcardSetDtoToEntity(flashcardSetInput);
+        log.debug("Mapped entity: " + mappedEntity);
```
Comment (PaulBredl) [inline src/main/java/de/unistuttgart/iste/gits/flashcard_service/service/FlashcardService.java]: TODO: remove all debug logs

**Review by PaulBredl — APPROVED** [review]
As discussed, there are still todos, I collected them, but should be merged so the frontend can work on it

**Comment by PaulBredl** [comment]
@ZombieAlienRobot the docker setup and stuff is definitely not correct, but I didn't know what exactly needs to be done. Maybe you want to have a look?

File: docker-compose.yml (hunk 24 above)
Comment (ZombieAlienRobot) [reply docker-compose.yml]: The network stuff doesn't need to be changed at all, just keep it the way it is/revert it to how it was

File: docker-compose.yml [hunk 33]
```diff
@@ -1,74 +1,75 @@
-version: "3"
```
Comment (ZombieAlienRobot) [inline docker-compose.yml:1]: don't remove the version here as some of the features we use require version 3

Comment (Keyshal-211) [reply docker-compose.yml:1]: I added

File: src/main/java/de/unistuttgart/iste/gits/flashcard_service/FlashcardServiceApplication.java (hunk 25 above)
Comment (Keyshal-211) [reply src/main/java/de/unistuttgart/iste/gits/flashcard_service/FlashcardServiceApplication.java]: done
